core/templates/dev/head/pages/exploration_editor/feedback_tab/ThreadDataService.js

```diff
@@ -80,15 +80,13 @@ oppia.factory('ThreadDataService', [
             $log.error('Number of suggestion threads doesn\'t match number of' +
                        'suggestion objects');
           }
-          for (var i = 0; i < res[1].data.suggestions.length; i++) {
-            var suggestion = SuggestionObjectFactory.createFromBackendDict(
-              res[1].data.suggestions[i]);
-            for (var j = 0; j < suggestionThreads.length; j++) {
-              if (suggestion.threadId() ===
-                  suggestionThreads[j].thread_id) {
+          for (var i = 0; i < suggestionThreads.length; i++) {
+            for (var j = 0; j < res[1].data.suggestions.length; j++) {
+              if (suggestionThreads[i].thread_id ===
+                  res[1].data.suggestions[j].threadId()) {
```

Comment:
I don't think I understand what's going on with this change...

In the old version, we have SuggestionObjectFactory.createFromBackendDict(res[1].data.suggestions[i]) and we call threadId() on that (which really should be getThreadId() btw).

In the new version, we have res[1].data.suggestions[j].threadId() directly.

Shouldn't one of these fail badly? If so why wasn't this caught?